spec/models/user_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  context 'testing user with no values' do
+    subject { User.new }
+    before { subject.save }
+
+    it 'require user to have name' do
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts counter should be an integer and greater than or equal to zero' do
+      expect(subject.posts_counter).to be_a(Integer)
+      expect(subject.posts_counter).to be >= 0
+    end
+
+    it 'return return_three_most_recent_posts' do
+      expect(subject.return_three_most_recent_posts).to eq(subject.posts.order(created_at: :desc).limit(3))
+    end
+  end
+  context 'testing user with values' do
+    subject(:user) { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+    before { user.save }
+
+    it 'user with name should be valid' do
+      expect(user).to be_valid
+    end
+
+    it 'posts counter should be equal to 0' do
+      expect(user.posts_counter).to eq(0)
+    end
+  end
```

Comment:
- [ ] The following test cases failed, kindly write test that passes. 
![image](https://github.com/Victoria129/blog-app-rails/assets/84156714/e3c8774c-6c23-4aa0-b03c-0a0ac0280b2e)

The images below show specific issues that caused the failure. Please read them carefully and fix.
![image](https://github.com/Victoria129/blog-app-rails/assets/84156714/50492277-f3ba-41c1-ac79-ea56ee676713)

![image](https://github.com/Victoria129/blog-app-rails/assets/84156714/da0b1ee5-ef34-455e-8cc7-9f5edefa7221)

![image](https://github.com/Victoria129/blog-app-rails/assets/84156714/b7e78dd9-5c85-4754-93aa-5f3dab6acde7)

![image](https://github.com/Victoria129/blog-app-rails/assets/84156714/e270c2c0-f2c8-47a5-ad12-aaae15024b6e)